rust/lance/src/index.rs

```diff
@@ -554,6 +556,17 @@ impl DatasetIndexInternalExt for Dataset {
                 .await
             }
 
+            (0, 3) => {
```

Comment:
do we need to bump versino everytime a new index is added?

Reply:
the new index is based on v2 lance format, which is not compatible with v1, so have to bump version